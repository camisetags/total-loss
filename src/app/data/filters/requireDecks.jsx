import * as React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { object, arrayOf, shape } from 'prop-types';

const DeckFilter = (NextComponent) => {
  const Filter = (props) => {
    if (props.list.length <= 1 && props.selected.cards.length === 0) {
      return <Redirect to="deck-select" />;
    }

    return <NextComponent {...props} />;
  };

  Filter.propTypes = {
    selected: shape({
      cards: arrayOf(object).isRequired,
    }).isRequired,
    list: arrayOf(object).isRequired,
  };

  const mapStateToProps = state => ({
    selected: state.deck.selected,
    list: state.deck.list,
  });

  /* istanbul ignore next */
  if (process.env.NODE_ENV === 'test') {
    return Filter;
  }

  /* istanbul ignore next */
  return connect(mapStateToProps)(Filter);
};

export default DeckFilter;
