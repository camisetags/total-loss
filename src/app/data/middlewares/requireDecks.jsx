/* eslint no-console: 0 */
// @flow
import * as React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import type { DeckState } from '../deck/state';
import type { RootState } from '../types';

const DeckFilter = (NextComponent: any): any => {
  const Filter = (props: DeckState): React.Element<*> => {
    if (props.list.length <= 1 && props.selected.cards.length === 0) {
      return <Redirect to="deck-select" />;
    }

    return <NextComponent {...props} />;
  };

  const mapStateToProps: Object = (state: RootState) => ({
    selected: state.deck.selected,
    list: state.deck.list,
  });

  return connect(mapStateToProps)(Filter);
};

export default DeckFilter;
