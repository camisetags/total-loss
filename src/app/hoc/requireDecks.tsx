import * as React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { RootState } from '../data/types';
// import * as deckRepository from '../deck/repository';

type FilterProps = {
  decks: object[];
};

const DeckFilter = (NextComponent: React.ReactType) => {
  const Filter = (props: FilterProps) => {
    if (props.decks.length <= 1) {
      return <Redirect to="deck-select" />;
    }

    return <NextComponent {...props} />;
  };

  const mapStateToProps = (state: any) => ({
    decks: Object.values(state.deck),
  });

  /* istanbul ignore next */
  if (process.env.NODE_ENV === 'test') {
    return Filter;
  }

  /* istanbul ignore next */
  return connect(mapStateToProps)(Filter);
};

export default DeckFilter;
