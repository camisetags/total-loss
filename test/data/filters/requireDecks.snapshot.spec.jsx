/* eslint react/prop-types:0 */
import React from 'react';
import { Redirect } from 'react-router-dom';
import renderer from 'react-test-renderer';
import requireDecks from '../../../src/app/data/filters/requireDecks';

describe('requireDecks middleware', () => {
  const simpleCompoenent = props => (
    <div>
      Simple Compoenent
      {props.selected.name}
    </div>
  );

  it('should return RedirectComponent', () => {
    const deckState = {
      selected: {
        name: '',
        description: '',
        cards: [],
      },

      list: [
        {
          name: '',
          description: '',
        },
      ],
    };
    const filter = requireDecks(simpleCompoenent)(deckState);

    expect(JSON.stringify(filter)).toEqual(JSON.stringify(<Redirect to="deck-select" />));
  });

  it('should return NextComponent', () => {
    const deckState = {
      selected: {
        name: '',
        description: '',
        cards: [],
      },

      list: [
        {
          name: '',
          description: '',
        },
        {
          name: '',
          description: '',
        },
      ],
    };

    const filter = requireDecks(simpleCompoenent)(deckState);
    const rendered = renderer.create(filter);

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
