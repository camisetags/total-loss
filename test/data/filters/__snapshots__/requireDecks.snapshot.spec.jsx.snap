// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requireDecks middleware should return NextComponent 1`] = `
<div>
  Simple Compoenent
  
</div>
`;
